test/client-schema-does-not-change-on-request-test.js

```diff
@@ -0,0 +1,125 @@
+
+      if(!requestContext.responseToSend)return requestContext.doneHandler();
+      if(requestContext.responseHttpHeaders){
+        for(const headerKey in requestContext.responseHttpHeaders ){
+          res.setHeader(headerKey,requestContext.responseHttpHeaders[headerKey]);
+        }
+      }
+      res.end(requestContext.responseToSend);
+
+      requestContext.expectedRequest = null;
+      requestContext.responseToSend = null;
+    });
+  }
+};
+
+describe("SOAP Client schema does not change", () => {
+
+  before(function(done){
+    server = http.createServer(requestContext.requestHandler);
+    server.listen(0, function(e){
+      if(e)return done(e);
+      port = server.address().port;
+      done();
+    });
+  });
+  it("should not change the schema", (done) => {
+
+    const tpath = path.join(__dirname, 'request-response-samples', "RetrieveFareQuoteDateRange__should_handle_child_namespaces");
+    const wsdlPath = path.resolve(tpath, "soap.wsdl");
+    const requestJSON = require(path.resolve(tpath, "request.json"));
+    const requestXML = fs.readFileSync(path.resolve(tpath, "request.xml"), {encoding: 'utf8'});
+    const responseJSON = require(path.resolve(tpath, "response.json"));
+    const responseXML = fs.readFileSync(path.resolve(tpath, "response.xml"), {encoding: 'utf8'});
+    const methodName = 'RetrieveFareQuoteDateRange';
+
+    requestContext.expectedRequest = requestXML;
+    requestContext.responseToSend = responseXML;
+    soap.createClient(wsdlPath, {disableCache: true}, function(err, client){
+        if (err) {
+          throw err;
+        }
+        //throw more meaningful error
+        if(typeof client[methodName] !== 'function'){
+          throw new Error('method ' + methodName + ' does not exists in wsdl specified in test wsdl: ' + wsdlPath);
+        }
+        const typeBefore = client?.wsdl?.definitions?.schemas?.['http://tempuri.org/Service/Request']?.complexTypes?.TransactionInfo?.children?.[0]?.children?.[2]?.$type;
+        
+        cbCaller(client, methodName, requestJSON, responseJSON, null, {}, null, done);
+        
+        const typeAfter = client?.wsdl?.definitions?.schemas?.['http://tempuri.org/Service/Request']?.complexTypes?.TransactionInfo?.children?.[0]?.children?.[2]?.$type;
+        assert.equal(typeBefore, typeAfter);
+        
+      }, 'http://localhost:'+port+'/Message/Message.dll?Handler=Default');
+  });
+});
+
+function cbCaller(client, methodName, requestJSON, responseJSON, responseSoapHeaderJSON, options, attachmentParts, done){
+    client[methodName](requestJSON, function(err, json, body, soapHeader){
+      if(requestJSON){
+        if (err) {
+          assert.notEqual('undefined: undefined', err.message);
+          assert.deepEqual(err.root, responseJSON);
+        } else {
+          // assert.deepEqual(json, responseJSON);
+          assert.equal(JSON.stringify(typeof json === 'undefined' ? null : json), JSON.stringify(responseJSON));
```

Comment:
I am wondering what was the reason to use stringify and equal instead of deepEqueal?

Reply:
That's shamelessly copied from the request/response tests...

Reply:
Does it work with the deepEqual?

Reply:
done